src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -800,6 +800,16 @@ public override void OnSnapshotInstalled(TransactionOperationContext context, lo
                         onDatabaseChanged.Invoke(this, (db, lastIncludedIndex, "SnapshotInstalled"));
                 }, null);
             }
+
+            // If there are trusted certificates in the local state, we will register them in the cluster now
+            foreach (var localCertKey in GetCertificateKeysFromLocalState(context))
```

Comment:
No, you cannot do that. You need to do it the _other way around_.
Scan the certificates in the snapshot and delete all the matching local ones.